### DISCUSSION TITLE
Pin npm dependency to certain major

### DISCUSSION THREAD

**Discussion by FelixBarsnickTS** [renovatebot/renovate#15564] [Request Help] [answered]
Hi everyone,

I have a problem pinning dependencies to a certain major version. Nodejs and the typescript dependency @types/node should be on the same major version, in my case its lts v14.
It works for the .nvmrc (node version manager) file and updates it to the newest minor and patch version. But for the dependency itself in the package.json it just updates to 14.0.0 but not to the newest minor/patch.

### My config

The first config also extends part 2 and 3. Due to company policies I am not able to post the entire files.

#### part 1
"$schema": "https://docs.renovatebot.com/renovate-schema.json",
"extends": [ "config:base" ]
#### part 2
"packageRules": [{
    "matchUpdateTypes": ["major"],
    "addLabels": ["major"]
  }, {
    "matchManagers": ["npm"],
    "matchUpdateTypes": ["minor", "patch", "pin", "digest", "lockFileMaintenance", "rollback", "bump"],
    "addLabels": ["minor"],
    "groupName": "minor - npm"
  } ]
#### part 3
"pinVersions": false,
  "ignoreDeps": ["asdf"],
  "packageRules": [
    { "matchPackageNames": ["node"],        "allowedVersions": "<=14", "description": "keep .nvmrc at same major as nodejs" },
    { "matchPackageNames": ["cimg/node"],   "allowedVersions": "<=14", "description": "keep circleci docker image at same major as nodejs" },
    { "matchPackageNames": ["@types/node"], "allowedVersions": "<=14", "description": "keep npm package at same major as nodejs" }
  ],
  "ignorePresets": ["workarounds:typesNodeVersioning"]

### Expected behavior

I would like to pin @types/node to the newest version of major 14. In the debug log I can see that renovate is aware of the newest minor and patch version but still sets the PR to version 14.0.0

#### debug excerpt

{
                 "depType": "devDependencies",
                 "depName": "@types/node",
                 "currentValue": "^14.0.0",
                 "datasource": "npm",
                 "prettyDepType": "devDependency",
                 "lockedVersion": "14.18.16",
                 "depIndex": 35,
                 "updates": [
                   {
                     "bucket": "major",
                     "newVersion": "17.0.32",
                     "newValue": "^17.0.0",
                     "releaseTimestamp": "2022-05-10T21:32:12.360Z",
                     "newMajor": 17,
                     "newMinor": 0,
                     "updateType": "major",
                     "isRange": true,
                     "branchName": "renovate/node-17.x"
                   }
                 ],

**Comment by viceice** [comment, marked as answer]
by default renovate won't bump ranges. checkout range strategy in the docs

**Comment by FelixBarsnickTS** [comment]
Thank you so very much, that solves my problem after weeks of trying stuff!

**Reply by glensc** [reply]
this is the documentation for range strategy:
- https://docs.renovatebot.com/configuration-options/#rangestrategy

but what is the final outcome to put to renovate.json to achieve "node: ^16" in package.json, and renovate updating only lock file with 16.9, 16.10, ...?

**Reply by rarkins** [reply]
Please start a new discussion and elaborate more on what you mean (I don't understand the above)

**Reply by glensc** [reply]
what was the solution for original reporter?

**Reply by glensc** [reply]
but i had similar question formed new discussion:
- https://github.com/renovatebot/renovate/discussions/20927
